py_zipkin/logging_helper.py

```diff
@@ -159,17 +167,33 @@ def log_spans(self):
                 self.thrift_endpoint,
             )
 
+            if self.report_root_timestamp:
+                timestamp = self.start_timestamp
+                duration = server_end_time - self.start_timestamp
+            else:
```

Comment:
yep in the brave code, looks similar, too, except it uses two signals: shared (ex I didn't start a new trace from an incoming request) and "sr"

```java
    // don't report server-side timestamp on shared or incomplete spans
    if (shared && (flags & FLAG_SR) != 0) {
      span.timestamp(null).duration(null);
    }
```